### ISSUE TITLE
Using pagerduty_alert_grouping_setting causes error `HTTP response failed with status code 404 and no JSON error object was present`

### ISSUE THREAD

**Issue by yeungalan0** [PagerDuty/terraform-provider-pagerduty#944]
Hello,

Thanks for the great work you all have put into this module!

### Terraform Version

v1.9.8

### Affected Resource(s)
- pagerduty_alert_grouping_setting

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration Files
```hcl
resource "pagerduty_service" "service" {
  name                    = var.service_name
  description             = var.service_description
  auto_resolve_timeout    = "null"
  acknowledgement_timeout = 1800
  escalation_policy       = var.escalation_policy_id
  alert_creation          = "create_alerts_and_incidents"

  auto_pause_notifications_parameters {
    enabled = true
    timeout = 120
  }

  incident_urgency_rule {
    type    = "constant"
    urgency = var.service_urgency
  }
}

resource "pagerduty_alert_grouping_setting" "service_grouping" {
  name    = "${var.service_name} Grouping"
  services = [pagerduty_service.service.id]
  type    = "intelligent"
  config {
    time_window = 300
  }
}
```

### Expected Behavior
pagerduty_alert_grouping_setting resource successfully created.

### Actual Behavior
I get the below error:

```
module.infra.module.pagerduty.pagerduty_alert_grouping_setting.service_grouping: Creating...
╷
│ Error: Unable to obtain list of alert grouping settings
│ 
│   with module.infra.module.pagerduty.pagerduty_alert_grouping_setting.service_grouping,
│   on ../../infra/pagerduty/service.tf line 32, in resource "pagerduty_alert_grouping_setting" "service_grouping":
│   32: resource "pagerduty_alert_grouping_setting" "service_grouping" {
│ 
│ HTTP response failed with status code 404 and no JSON error object was present
```

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

**Comment by alxndr13** [comment]
Same error here, TF Version `1.9.5` and Provider Version `3.17`. 

As this worked for us last week, it seems like something on the backend side of things changed.

**Comment by alxndr13** [comment]
@yeungalan0 fixed that in #951, if you want to locally override your provider to get the change through.

**Comment by alxndr13** [comment]
Fixed in the library: https://github.com/PagerDuty/go-pagerduty/pull/545

**Comment by imjaroiswebdev** [comment]
Hi @yeungalan0 this issue should have been patched by version [v3.17.2](https://github.com/PagerDuty/terraform-provider-pagerduty/releases/tag/v3.17.2), thanks to @alxndr13 whom published the patch in our API client. Thank you both, for reporting and taking the time for providing a fix 🏆

**Comment by yeungalan0** [comment]
I just updated and can confirm this is working now. Thank you for the quick fix! 🙏
